src/corehost/cli/apphost/bundle/extractor.cpp

```diff
@@ -0,0 +1,150 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "runner.h"
+#include "extractor.h"
+#include "pal.h"
+#include "trace.h"
+#include "utils.h"
+
+using namespace bundle;
+
+// Compute the final extraction location as:
+// m_extraction_dir = $DOTNET_BUNDLE_EXTRACT_BASE_DIR/<app>/<id>/...
+//
+// If DOTNET_BUNDLE_EXTRACT_BASE_DIR is not set in the environment, the 
+// base directory defaults to $TMPDIR/.net
+void extractor_t::determine_extraction_dir()
+{
+    if (!pal::getenv(_X("DOTNET_BUNDLE_EXTRACT_BASE_DIR"), &m_extraction_dir))
+    {
+        if (!pal::get_temp_directory(m_extraction_dir))
+        {
+            trace::error(_X("Failure processing application bundle."));
+            trace::error(_X("Failed to determine location for extracting embedded files"));
+            throw StatusCode::BundleExtractionFailure;
+        }
+
+        append_path(&m_extraction_dir, _X(".net"));
```

Comment:
In .NET Core we typically use `dotnet` as the name which refers to the product when used by file/directory names. Maybe I'm biased, but if I would see `.net` I would think .NET Framework.

Reply:
The directory to be extracted to is a topic of ongoing discussion; and I would like to address it separately. 

This PR doesn't change the extraction directory, it keeps the extraction location as described in the [design](https://github.com/dotnet/designs/blob/master/accepted/single-file/extract.md).